sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -775,6 +778,57 @@ def _get_sample_count(self, n_samples):
             return ceil(n_samples * self.subsample)
         return n_samples
 
+    def _get_color_dict_from_values(
+        self, value_list, categorical_or_continuous: str, cmap_name: str = "Spectral"
+    ):
+        color_dict = {}
+        if categorical_or_continuous == "categorical":
+            mapping = {
+                np.unique(value_list)[i]: f"C{i}"
+                for i in range(len(np.unique(value_list)))
+            }
+            color_dict["color_list"] = [
+                mapping[value_list[i]] for i in range(len(value_list))
+            ]
+            color_dict["categorical_mapping"] = mapping
+        elif categorical_or_continuous == "continuous":
+            cmap = mpl.cm.get_cmap(cmap_name)
+            norm = mpl.colors.Normalize(vmin=value_list.min(), vmax=value_list.max())
+            color_dict["color_list"] = [
+                mpl.colors.to_hex(cmap(norm(value))) for value in value_list
+            ]
+            color_dict["norm"] = norm
+            color_dict["cmap"] = cmap
+        return color_dict
+
+    def _check_if_colors_by_feature(self, ice_lines_kw, preds):
```

Comment:
der name ist etwas irreführend: bei einem "check" würde ich erwarten, dass die funktion ein true oder false zurück liefert, nicht die fertigen farben^^

Reply:
wenn du eh nur `preds.shape[0]` brauchst würde ich auch nur das als `n` übergeben

Reply:
bzw schau nochmal ob das nicht iwo in der klasse sogar abgespeichert ist iwo, also ob du auf iwas von `self` zugreifen kannst, was auch die info enthält